### PR TITLE
Use setInterval to upload metrics to Stackdriver periodically

### PR DESCRIPTION
**Pull request by rueian**
Hi, 

I made some changes to let stackdriver monitoring exporter upload metrics periodically, which are related to issue #152. Including:

1. Use `setInterval()` to schedule metrics uploading.
2. Add the `close()` method which can clear the setInterval timer.
3. Change the behavior of `.onRecord()` and related test cases, because metrics will not be uploaded when `.onRecord()` is called.
4. Make sure `.catch()` blocks in some test cases are executed by finishing them with mocha `done()`
5. Map all non `AggregationType.LAST_VALUE` to stackdriver `MetricKind.CUMULATIVE`.

### CODE REVIEW

File: packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts [hunk 1]
```diff
@@ -70,13 +87,28 @@ export class StackdriverStatsExporter implements StatsEventListener {
...
+      const tagsKey = this.encodeTags(measurement.tags);
+      this.viewToUpload[view.name].tags[tagsKey] = measurement.tags;
+    }
+  }
 
-    const timeSeries = views.map(view => {
-      return this.createTimeSeriesData(view, measurement);
-    });
+  async uploadViews() {
+    const timeSeries = [] as TimeSeries[];
```
Comment (kjin) [inline packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts]:
```suggestion
    const timeSeries: TimeSeries[] = [];
```

File: packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts [hunk 2]
```diff
@@ -70,13 +87,28 @@ export class StackdriverStatsExporter implements StatsEventListener {
    * @param measurement The measurement recorded
    */
   async onRecord(views: View[], measurement: Measurement) {
```
Comment (kjin) [inline packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts]:
```suggestion
  onRecord(views: View[], measurement: Measurement) {
```

File: packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts [hunk 3]
```diff
@@ -70,13 +87,28 @@ export class StackdriverStatsExporter implements StatsEventListener {
...
+      }
+      const tagsKey = this.encodeTags(measurement.tags);
+      this.viewToUpload[view.name].tags[tagsKey] = measurement.tags;
+    }
+  }
 
-    const timeSeries = views.map(view => {
-      return this.createTimeSeriesData(view, measurement);
-    });
+  async uploadViews() {
```
Comment (kjin) [inline packages/opencensus-exporter-stackdriver/src/stackdriver-monitoring.ts]:
```suggestion
 private uploadViews() {
```

File: packages/opencensus-exporter-stackdriver/src/types.ts [hunk 4]
```diff
@@ -47,6 +47,8 @@ export interface StackdriverExporterOptions extends ExporterConfig {
    * of a stackdriver metric. Optional
    */
   metricPrefix?: string;
+
+  onMetricUploadError?: (err: Error) => void;
```
Comment (kjin) [inline packages/opencensus-exporter-stackdriver/src/types.ts]: Can you add docs for this field?

File: packages/opencensus-exporter-stackdriver/test/test-stackdriver-monitoring.ts [hunk 5]
```diff
@@ -329,41 +362,57 @@ describe('Stackdriver Stats Exporter', function() {
...
+              return assertTimeSeries(
+                  exporterTestLogger.debugBuffer[0][0], viewTimeSeries,
+                  measurement, PROJECT_ID, prefixExporterOptions.metricPrefix);
+            });
       });
     });
 
     describe('With no network connection', () => {
-      it('.onRegisterView() Should fail by network error', async () => {
+      it('.onRegisterView() Should fail by network error', (done) => {
```
Comment (kjin) [inline packages/opencensus-exporter-stackdriver/test/test-stackdriver-monitoring.ts]: Why was this changed from being an async function?

Comment (kjin) [reply packages/opencensus-exporter-stackdriver/test/test-stackdriver-monitoring.ts]: I see, good catch. Thanks!

**Comment by kjin** [comment]
@isaikevych please merge at your leisure.

@rueian thanks for the contribution!

**Comment by isaikevych** [comment]
Fixes https://github.com/census-instrumentation/opencensus-node/issues/152
